src/v/pandaproxy/sharded_client_cache.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+#include "config/broker_authn_endpoint.h"
+#include "config/rest_authn_endpoint.h"
+#include "pandaproxy/types.h"
+
+#include <seastar/core/sharded.hh>
+
+#include <vector>
+
+namespace pandaproxy {
+
+class kafka_client_cache;
+
+class sharded_client_cache {
+public:
+    ss::future<> start(
+      ss::smp_service_group sg,
+      YAML::Node const& cfg,
+      size_t size,
+      std::vector<config::broker_authn_endpoint> kafka_api,
+      model::timestamp::type keep_alive = 30000);
```

Comment:
I don't understand your statement. Is it not already configurable?

Reply:
You are right, of course. I don't think the default is necessary, it would be preferable to always wire up the config value (and make it updatable without a restart, if it's a cluster level config).

Reply:
It is not a customer facing config but instead something that we control internally.
I originally had keep_alive in the config file but some RFC comments advised against this for now.

Reply:
I'm going to make this a config option on pandaproxy_api. It'll be easier to tune keep alive then and I can make it `needs_restart::no`